packages/cli/package/test/tests/provider.test.ts

```diff
@@ -165,6 +207,452 @@ describe("provider tests", () => {
+        ...STORAGE_RESOURCE_METADATA,
+        type: "ODD",
+      };
+
+      const STORAGE_RESOURCE_NAME_UPDATED =
+        storageResourceToHumanReadableString(STORAGE_RESOURCE_METADATA_UPDATED);
+
+      const BANDWIDTH_RESOURCE_METADATA_UPDATED: BandwidthMetadata = {
+        ...BANDWIDTH_RESOURCE_METADATA,
+        type: "semi-shared",
+      };
+
+      const BANDWIDTH_RESOURCE_NAME_UPDATED =
+        bandwidthResourceToHumanReadableString(
+          BANDWIDTH_RESOURCE_METADATA_UPDATED,
+        );
+
+      const IP_RESOURCE_METADATA_UPDATED: IPMetadata = {
+        ...IP_RESOURCE_METADATA,
+        version: "10",
+      };
+
+      const IP_RESOURCE_NAME_UPDATED = ipResourceToHumanReadableString(
+        IP_RESOURCE_METADATA_UPDATED,
+      );
+
+      await signBatch({
+        title: "Update resources metadata",
+        populatedTxs: [
+          populateTx(
+            contracts.diamond.updateResourceMetadata,
+            cpuResourceId,
+            JSON.stringify(CPU_RESOURCE_METADATA_UPDATED),
+          ),
+          populateTx(
+            contracts.diamond.updateResourceMetadata,
+            ramResourceId,
+            JSON.stringify(RAM_RESOURCE_METADATA_UPDATED),
+          ),
+          populateTx(
+            contracts.diamond.updateResourceMetadata,
+            storageResourceId,
+            JSON.stringify(STORAGE_RESOURCE_METADATA_UPDATED),
+          ),
+          populateTx(
+            contracts.diamond.updateResourceMetadata,
+            bandwidthResourceId,
+            JSON.stringify(BANDWIDTH_RESOURCE_METADATA_UPDATED),
+          ),
+          populateTx(
+            contracts.diamond.updateResourceMetadata,
+            ipResourceId,
+            JSON.stringify(IP_RESOURCE_METADATA_UPDATED),
+          ),
+        ],
+        providerOrWallet,
+      });
+
+      await expect(async () => {
+        await fluence({
+          args: ["provider", "gen"],
+          flags: PRIV_KEY_1,
+          cwd,
+        });
+      }).rejects.toThrow();
+
+      setResourceNamesInConfig({
+        cpuName: CPU_RESOURCE_NAME_UPDATED,
+        ramName: RAM_RESOURCE_NAME_UPDATED,
+        storageName: STORAGE_RESOURCE_NAME_UPDATED,
+        bandwidthName: BANDWIDTH_RESOURCE_NAME_UPDATED,
+        ipName: IP_RESOURCE_NAME_UPDATED,
+      });
+
+      await writeFile(
+        providerConfig.$getPath(),
+        stringify(
+          removeProperties(providerConfig, ([, v]) => {
+            return typeof v === "function";
+          }),
+        ),
+        "utf8",
+      );
+
+      await fluence({
+        args: ["provider", "gen"],
+        flags: PRIV_KEY_1,
+        cwd,
+      });
+
+      const offerInfoWithUpdatedResources = await fluence({
+        args: ["provider", "offer-info"],
+        flags: {
+          ...PRIV_KEY_1,
+          [OFFER_FLAG_NAME]: defaultOfferName,
+        },
+        cwd,
+      });
+
+      expect(offerInfoWithUpdatedResources).toEqual(
```

Comment:
Maybe we need to check rest of updated values?

```
const updatedResourceTypes = [
    BANDWIDTH_RESOURCE_METADATA_UPDATED,
    CPU_RESOURCE_NAME_UPDATED,
    RAM_RESOURCE_NAME_UPDATED,
    STORAGE_RESOURCE_NAME_UPDATED,
    IP_RESOURCE_NAME_UPDATED,
  ].forEach(resource => {
    expect(offerInfoWithUpdatedResources).toEqual(
      expect.stringContaining(resource.type),
    );
  });
```

Reply:
yeah, will add something like this. Thanks